gtk+3 3.16.2 and gtkmm3 3.16.0

tschoonj opened the pull request:
version bumps
Due to the new dependency on libepoxy, Quartz has become the only
supported backend according to the discussion in
anholt/libepoxy#28 and this seems unlikely to
change in the future.

mikemcquaid commented on Library/Formula/gtk+3.rb:
```diff
@@ -37,21 +30,37 @@ def install
   end
 
   def post_install
     system "#{Formula["glib"].opt_bin}/glib-compile-schemas", "#{HOMEBREW_PREFIX}/share/glib-2.0/schemas"
```
This line needs fixed if there's no `glib` dependency any more.

tschoonj replied:
the glib dependency is still there, but through pango

mikemcquaid replied:
Would be good to declare it here too as we're using it in the formula

mikemcquaid commented on Library/Formula/gtk+3.rb:
```diff
@@ -37,21 +30,37 @@ def install
     system "make", "install"
     # Prevent a conflict between this and Gtk+2
     mv bin/"gtk-update-icon-cache", bin/"gtk3-update-icon-cache"
+    rm_f share/"glib-2.0/schemas/gschemas.compiled"
```
Why?

tschoonj replied:
this file is giving the buildbot headaches. It is generated by post_install in both gsettings-desktop-schemas and gtk+3 and is not removed when uninstalling these ports (as it should).

mikemcquaid replied:
Ok, thanks.

mikemcquaid commented on Library/Formula/gtk+3.rb:
```diff
@@ -37,21 +30,37 @@ def install
-    end
+    # --enable-x11-backend not possible because of libepoxy, which is now a hard dependency of gtk3.
+    # Looks like this will remain so
+    # according to the comments in https://github.com/anholt/libepoxy/pull/28
```
Can probably just make this a commit message.

mikemcquaid commented on Library/Formula/gtk+3.rb:
```diff
@@ -37,21 +30,37 @@ def install
 
     system "./configure", *args
+    # necessary to avoid gtk-update-icon-cache not being found during make install
+    mkdir_p bin
```
`bin.mkpath`

mikemcquaid commented on Library/Formula/gtk+3.rb:
```diff
@@ -37,21 +30,37 @@ def install
   def post_install
     system "#{Formula["glib"].opt_bin}/glib-compile-schemas", "#{HOMEBREW_PREFIX}/share/glib-2.0/schemas"
   end
+  test do
```
Add a newline before this.

tschoonj replied:
done

mikemcquaid commented on Library/Formula/gsettings-desktop-schemas.rb:
```diff
@@ -26,6 +26,7 @@ def install
+    rm_f share/"glib-2.0/schemas/gschemas.compiled"
   end
 
   def post_install
```
Looks like this is failing, not sure if it's related: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/24793/version=yosemite/testReport/junit/brew-test-bot/yosemite/install___gsettings_desktop_schemas_3_14_0_yosemite_bottle_2_tar_gz/

tschoonj commented:
Any idea what's causing this problem here?

tschoonj commented:
Ok, changed the post_install message in gsettings-desktop-schemas to match the one in gtk+3. Let's see if this one works...

tschoonj commented:
Also seems to fail... Is there any way we can get more information from the buildbot?

tschoonj commented:
New attempt: glib is now a run-time requirement...

tschoonj commented:
Yes! So that was the problem...

tschoonj commented:
I removed the rm_f line in gtk+3, I should have done it before as it was not causing the issue with gsettings-desktop-schemas.

I also added the glib dependency again though it's useless: it's already a runtime dependency through pango. If this was not the case then the buildbot would already have generated the same error as for gsettings-desktop-schemas

mikemcquaid commented:
> I also added the glib dependency again though it's useless: it's already a runtime dependency through pango. If this was not the case then the buildbot would already have generated the same error as for gsettings-desktop-schemas

It's not useless, the way our buildsystem handles things every formula should declare all it's dependencies rather than relying on dependencies to pull in their own dependencies. Thanks!

tschoonj commented:
Ok, done!

What about #38969? Will you pull this one in or will you wait until you decide to go all-quartz? The build failed but seems due do to an issue with the mountain lion buildbot

mikemcquaid commented:
> What about #38969? Will you pull this one in or will you wait until you decide to go all-quartz?

Let's wait until we go all-Quartz.

tschoonj commented:
Ok, will change the formulas then if you decide to do so.



Op 24-apr.-2015 om 18:22 heeft Mike McQuaid <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


What about #38969<https://github.com/Homebrew/homebrew/pull/38969>? Will you pull this one in or will you wait until you decide to go all-quartz?

Let's wait until we go all-Quartz.

—
Reply to this email directly or view it on GitHub<https://github.com/Homebrew/homebrew/pull/38972#issuecomment-95983987>.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

tschoonj commented:
Did you already merge this one in? I see no reference to it in the commit history...

Also could you pull in #38941? Thanks!

mikemcquaid commented:
@tschoonj Sorry, thought I had but missed it. Done now!
